elixir/apps/web/lib/web/components/navigation_components.ex

```diff
@@ -154,6 +156,7 @@ defmodule Web.NavigationComponents do
       flex items-center p-2
       text-base font-medium text-gray-900
       rounded-lg
+      #{if String.starts_with?(@active_path, @navigate), do: @active_class, else: ""}
       hover:bg-gray-100
       dark:text-white dark:hover:bg-gray-700 group]}>
```

Comment:
I think for state changes we should move them out of `~H` and do something like:
```
activity_class = 
  if String.starts_with?(assigns.current_path, assigns.navigate) do
    assigns.active_class
  end

assigns = assign(assigns, activity_class: activity_class)
```

And then to make classes more readable we can leverage list of strings instead of sigil (I noticed Elixir formatted makes some weird moves with it so was trying to avoid it in new code):
```
~H"""
... class={[
  "flex items-center p-2",
  "text-base font-medium text-gray-900",
  "rounded-lg",
  @activity_class
]}
"""
```

Reply:
👍 Will get this feedback applied

Reply:
Or we can do like Phoenix team does:
```
class={[
  "flex items-center p-2",
  "text-base font-medium text-gray-900",
  "rounded-lg",
  String.starts_with?(assigns.current_path, assigns.navigate) && @active_class,
]}
```

Reply:
I think the second one might end up being clean because we need the particular `navigate` attribute for each item inside the comprehension

Reply:
Updated